providers/mount.rb

```diff
@@ -81,11 +81,11 @@ def basic_mount_options
 end
 
 def mount_options_for_backup_server
-  case new_resource.backup_server.class
+  case new_resource.backup_server
   when String
     ',backupvolfile-server=' + new_resource.backup_server
   when Array
-    ',backupvolfile-server=' + new_resource.backup_server.join(',backupvolfile-server=')
+    ',backup-volfile-servers=' + new_resource.backup_server.join(':')
```

Comment:
It does look like you are correct though
https://github.com/gluster/glusterfs/blob/2e23c62cc50037c8e61bcd9c04348409e7627181/xlators/mount/fuse/utils/mount_glusterfs.in#L266